samples/javascript/web/inventory/controllers/api.js

```diff
@@ -0,0 +1,103 @@
+const apiCookie = 'api-session';
+const hostCookie = 'host';
+
+// DEMO: Turn on SSL cert verification
+const useSSL = false;
+
+/**
+ * Process the POST request from the login page using the credentials to log into the vAPI endpoint.
+ * Upon login, redirects to /api displaying the available hosts on the vSphere instance.
+ */
+exports.postLogin = function(req, res, next) {
+  request = require('request');
+  
+  request({
+    url : req.body.host + process.env.LOGIN_PATH,
+    method: 'POST',
+    strictSSL: useSSL,
+    headers : {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        "Authorization" : "Basic " + new Buffer(req.body.user + ":" + req.body.password).toString("base64")
+    }
+  }, function(error, response, body) {
+    if (response.statusCode < 400) {
+      // Save the vmware-api-session and host to cookies on the client
+      if (response.headers['set-cookie'] && response.headers['set-cookie'][0].startsWith('vmware-api-session')) {
+        res.cookie(apiCookie, response.headers['set-cookie'][0], { maxAge: 900000, httpOnly: true });
+        res.cookie(hostCookie, req.body.host, { maxAge: 900000, httpOnly: true });      
+      }
+      // Now that we're authenticated render the API page
+      res.redirect('/inventory');
+    } else {
+      res.redirect('/');
+    }
+  });
+}
+
+/**
+ * Handles vSphere REST API requests and by default returns /vcenter/host results. If a "path" query param
+ * is provided it will be used to call the endpoint with that route returning the results. In the event
+ * the user is not logged in they are redirected to the home page. Only includes basic error handling.
+ */
+exports.getApi = async function(req, res, next) {
```

Comment:
The name `getApi` sounds a bit weird to me. I think `handleApiRequest` or something along those lines may be a little bit more helpful.